packages/rsocket-core/src/RequestChannelStream.ts

```diff
@@ -791,6 +793,8 @@ export class RequestChannelResponderStream
     Reassembler.cancel(this);
 
     if (!outboundDone) {
+      // TODO: receiver is not set when in constructed with follows flag.
+      //  should receiver always be set or should this check for null?
```

Comment:
As I called out in the PR description, I think there is either an issue with the assertions in various tests, or there is something wrong where we often do not have the expected state/value for `outboundDone`, and don't end up doing the expected side effects.